### DISCUSSION TITLE
Event propagation for focus and blur events

### DISCUSSION THREAD

**Discussion by yusufyildirim** [react-native-tvos/react-native-tvos#668] [Feature Requests]
Event propagation is a concept that is described here for those who are not familiar with it: https://javascript.info/bubbling-and-capturing

Most events in RN are `direct`, meaning, they do not propagate. Focus and blur events we have are also the same in `react-native-tvos`. However, it would be quite beneficial for certain cases if they were propagating.

We can listen bubbling focus events to trigger certain things, like animations, higher up in the tree somewhere. A common use-case would be animating side menus. There are bunch of other advanced use-cases too. One example being from this PR: https://github.com/facebook/react-native/pull/32646/files#diff-7481ec8ed5532798b075df637e805a8e439807aa2ce671208c24068c286361e8R2306 

We have a focus management logic inside VirtualizedList that always keep last focused item and its surrounding rendered. This is very much useful for a lot of scenarios. It depends on `onFocusCapture` event, though. It's not supported by `react-native-tvos` yet, so we're not able to leverage this optimization just yet.

I think we should do this. It's not super obvious how it should be implemented but I've managed to make it work for Android old arch. So it's very much supported. Things to consider;

We can provide capture events by default since they'll be a new. So, people should be able to start using `onFocusCapture` or `onBlurCapture` callbacks anywhere in the React tree to listen their descendant events.

Bubbling is a different story. People write their code with an assumption that focus/blur events won't bubble. So, it could potentially be a breaking change for some if we introduce bubbling by default. Instead, we should probably do that behind a feature flag.

**Comment by yusufyildirim** [comment]
PR's on the way: https://github.com/react-native-tvos/react-native-tvos/pull/813

**Comment by douglowder** [comment]
Feature is complete and shipped in 0.76.2-0: https://github.com/react-native-tvos/react-native-tvos/discussions/827
